src/main/java/com/flightstats/hub/model/ChannelConfig.java

```diff
@@ -93,15 +93,21 @@ private ChannelConfig(String name,
             this.ttlDays = ttlDays;
             this.maxItems = maxItems;
         }
-
         if (isBlank(storage)) {
-            this.storage = SINGLE.name();
+            this.storage = BATCH.name();
         } else {
             this.storage = StringUtils.upperCase(storage);
         }
 
+        if (isHistorical()) {
+            tags.add(HISTORICAL.toString());
+            if (StringUtils.isNotBlank(storage) && !storage.equals(SINGLE.name())) {
+                throw new InvalidRequestException(storage + " storage is not supported by the historical channel/mutable time api");
```

Comment:
It feels weird to have a constructor throw an `InvalidRequestException`. Is this class only used when creating new channels, or is it tied to deserializing state that's stored in dynamo? Is it possible this could blow up if there was some weird config stored in dynamo that shouldn't have been possible but the check didn't exist before?

Reply:
This check already happened, it happened at a different layer, in ChannelValidator which runs after this builder is built. So it ~~shouldn't~~ have been possible to have that state pre-existing in dynamo, I am working on an alternative strategy to throwing this right here because it is very gross